frontend/packages/operator-lifecycle-manager/src/components/descriptors/index.tsx

```diff
@@ -8,133 +8,202 @@ import { JSONSchema6 } from 'json-schema';
+            <div className="details-item__value--group">
+              <dl>
+                {_.map(arrayElementDescriptors, (primitiveDescriptor: Descriptor) => {
+                  const path = primitiveDescriptor.path.replace(/\d+/, String(i));
+                  return (
+                    <DescriptorDetailsItem
+                      descriptor={{ ...primitiveDescriptor, path }}
+                      key={`${type}.${path}`}
+                      model={model}
+                      obj={obj}
+                      onError={onError}
+                      schema={getSchemaAtPath(schema, path)}
+                      type={type}
+                    />
+                  );
+                })}
+              </dl>
+            </div>
+          ))
+        ) : (
+          <span className="text-muted">None</span>
+        )}
+      </div>
     </DetailsItem>
-  ) : null;
+  );
 };
 
-export const DescriptorDetailsItemList: React.FC<DescriptorDetailsItemListProps> = ({
-  descriptors,
-  itemClassName,
+const DescriptorDetailsItemGroup: React.FC<DescriptorDetailsItemGroupProps> = ({
+  group,
+  groupPath,
   model,
   obj,
   onError,
   schema,
   type,
 }) => {
-  const groupedDescriptors = (descriptors ?? []).reduce((acc, descriptor) => {
-    const [key] = _.toPath(descriptor.path);
-    return {
-      ...acc,
-      [key]: [...(acc?.[key] ?? []), descriptor],
-    };
-  }, {});
+  const { descriptor, nested } = group;
+  const groupSchema = getSchemaAtPath(schema, `${type}.${groupPath}`);
+  const description = descriptor?.description || groupSchema?.description;
+  const label = descriptor?.displayName || groupSchema?.title || _.startCase(groupPath);
+  const arrayGroups = _.pickBy(nested, 'isArrayGroup');
+  const primitives = _.omitBy(nested, 'isArrayGroup');
+  const className = _.isEmpty(arrayGroups) ? 'col-sm-6' : 'col-sm-12';
   return (
-    <dl className={`olm-descriptors olm-descriptors--${type}`}>
-      {_.map(groupedDescriptors, (group: Descriptor[], groupName) => (
-        <div key={`${type}.${groupName}`} className={itemClassName}>
-          {group.length > 1 ? (
-            <DescriptorDetailsItemGroup
-              descriptors={group}
-              groupName={groupName}
-              model={model}
-              obj={obj}
-              onError={onError}
-              schema={schema}
-              type={type}
-            />
-          ) : (
-            <DescriptorDetailsItem
-              descriptor={group[0]}
-              model={model}
-              obj={obj}
-              onError={onError}
-              schema={schema}
-              type={type}
-            />
-          )}
-        </div>
-      ))}
-    </dl>
+    <div className={className}>
+      <DetailsItem description={description} label={label} obj={obj} path={`${type}.${groupPath}`}>
+        <dl>
+          <div className="details-item__value--group">
+            <Flex justifyContent={{ default: 'justifyContentFlexStart' }}>
```

Comment:
Avoid `Flex` and `FlexItem`, which are essentially inline styles. Use BEM classes and CSS instead.